### ISSUE TITLE
[java] UnusedPrivateMethod reports a method is not used for .map(this::methodName)

### ISSUE THREAD

**Issue by mkleint** [pmd/pmd#4995]
**Affects PMD Version:**  7.1.0

**Rule:** UnusedPrivateMethod 

Please provide the rule name and a link to the rule documentation:
https://docs.pmd-code.org/pmd-doc-7.1.0/pmd_rules_java_bestpractices.html#unusedprivatemethod

**Description:**
see screenshot
<img width="1155" alt="Screenshot 2024-05-07 at 12 57 22 PM" src="https://github.com/pmd/pmd/assets/178549/4c4d031d-3704-4c7e-b888-603511b5a37b">

using java17 + maven pmd plugin 3.22.0

**Code Sample demonstrating the issue:**

```java

```

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** *[ Maven ]*

**Comment by jsotuyod** [comment]
This seems to be a duplicate of #4985.

This is due to a mistake in the way PMD is configured, namely, you should run the compiler before PMD is run in Maven. You should change the phase of execution of PMD to verify to make sure the compiler runs before PMD.

In the meantime #4989 (to be included in PMD 7.2.0) does improve on PMD's behavior around unknown types.
